orderer/consensus/smartbft/synchronizer_bft.go

```diff
@@ -0,0 +1,239 @@
+
+	bftDeliverer := &blocksprovider.BFTDeliverer{
+		ChannelID:                 s.Support.ChannelID(),
+		BlockHandler:              syncBuffer,
+		Ledger:                    &ledgerInfoAdapter{s.Support},
+		UpdatableBlockVerifier:    verifier,
+		Dialer:                    blocksprovider.DialerAdapter{ClientConfig: clientConfig},
+		OrderersSourceFactory:     &orderers.ConnectionSourceFactory{}, // no overrides in the orderer
+		CryptoProvider:            s.CryptoProvider,
+		DoneC:                     make(chan struct{}),
+		Signer:                    s.Support,
+		DeliverStreamer:           blocksprovider.DeliverAdapter{},
+		CensorshipDetectorFactory: &blocksprovider.BFTCensorshipMonitorFactory{},
+		Logger:                    flogging.MustGetLogger("orderer.blocksprovider").With("channel", s.Support.ChannelID()),
+		InitialRetryInterval:      10 * time.Millisecond, // TODO get it from config.
+		MaxRetryInterval:          2 * time.Second,       // TODO get it from config.
+		BlockCensorshipTimeout:    20 * time.Second,      // TODO get it from config.
+		MaxRetryDuration:          time.Minute,           // TODO get it from config.
+		MaxRetryDurationExceededHandler: func() (stopRetries bool) {
+			syncBuffer.Stop()
+			return true // In the orderer we must limit the time we try to do Synch()
+		},
+	}
+
+	s.Logger.Infof("Created a BFTDeliverer: %+v", bftDeliverer)
+	bftDeliverer.Initialize(lastConfigEnv.GetConfig(), myEndpoint)
+
+	go bftDeliverer.DeliverBlocks()
+	defer bftDeliverer.Stop()
+
+	//===
+	// Loop on sync-buffer
+
+	targetSeq := targetHeight - 1
+	seq := startHeight
+	var blocksFetched int
+
+	s.Logger.Debugf("Will fetch sequences [%d-%d]", seq, targetSeq)
+
+	var lastPulledBlock *common.Block
+	for seq <= targetSeq {
+		block := syncBuffer.PullBlock(seq)
+		if block == nil {
+			s.Logger.Debugf("Failed to fetch block [%d] from cluster", seq)
+			break
+		}
+		if protoutil.IsConfigBlock(block) {
+			s.Support.WriteConfigBlock(block, nil)
+		} else {
+			s.Support.WriteBlock(block, nil)
+		}
+		s.Logger.Debugf("Fetched and committed block [%d] from cluster", seq)
+		lastPulledBlock = block
+
+		prevInLatestDecision := s.lastReconfig.InLatestDecision
+		s.lastReconfig = s.OnCommit(lastPulledBlock)
+		s.lastReconfig.InLatestDecision = s.lastReconfig.InLatestDecision || prevInLatestDecision
+		seq++
+		blocksFetched++
+	}
+
+	syncBuffer.Stop()
+
+	if lastPulledBlock == nil {
+		return nil, errors.Errorf("failed pulling block %d", seq)
+	}
+
+	startSeq := startHeight
+	s.Logger.Infof("Finished synchronizing with cluster, fetched %d blocks, starting from block [%d], up until and including block [%d]",
+		blocksFetched, startSeq, lastPulledBlock.Header.Number)
+
+	viewMetadata, lastConfigSqn := s.getViewMetadataLastConfigSqnFromBlock(lastPulledBlock)
+
+	s.Logger.Infof("Returning view metadata of %v, lastConfigSeq %d", viewMetadata, lastConfigSqn)
+	return s.BlockToDecision(lastPulledBlock), nil
```

Comment:
why is `lastConfigSqn` interesting here?

Reply:
It is not...